rfcs/text/0049-entity-fields.md

```diff
@@ -20,31 +21,55 @@ This approach would allow ECS to accommodate new types of entities without requi
 
-When representing entities that correspond to existing ECS field sets (e.g., hosts, users, services, containers), the relevant ECS field set should be used to capture detailed metadata about that entity. For example:
+When representing entities that have existing ECS entiy field sets (e.g., hosts, users, services, containers), the relevant ECS field set should continue to be used to capture a copy of the detailed metadata about that entity. For example:
 
-- Host entities should utilize the `host.*` field set to capture detailed host information
-- User entities should leverage the `user.*` field set for user-specific attributes
-- Cloud resource entities should use the `cloud.*` field set for cloud provider details
-- Service entities should employ the `service.*` field set for service-specific metadata
+- Host entities should also utilize the `host.*` field set to capture detailed host information
+- User entities should also leverage the `user.*` field set for user-specific attributes
+- Cloud resource entities should also use the `cloud.*` field set for cloud provider details
+- Service entities should also employ the `service.*` field set for service-specific metadata
 
-This approach ensures backward compatibility, maintains existing ECS patterns, and preserves the rich metadata capabilities of established field sets while allowing the entity fields to provide a consistent way to identify and categorize all entity types. The entity fields serve as a complementary layer that enables unified entity representation, particularly for entity types that don't have dedicated field sets.
+This approach ensures backward compatibility, maintains existing ECS patterns, and preserves the rich metadata capabilities of established field sets while allowing the entity fields to provide a consistent way to identify and categorize *all* entity types. The entity fields serve as a complementary layer that enables unified entity representation, particularly for entity types that don't have dedicated field sets.
 
-## Usage
+## Field Re-use
 
-The entity field set enables us to normalize entity data in such a way where we can easily query key attributes in a standardized way regardless of the type and source of the entity. This will be how we'll normalize all entity data in the upcoming inventory experience that we're planning for the security solution.
+### ECS Producers
+
+For ECS producers, such as Beats, Elastic Agent integrations, ingest pipelines, and other methods for shipping data to Elastic, the `entity.*` fields are expected to be nested as follows:
+- If the entity type is one of host, user, service, cloud, orchestrator), then the entity fields should be nested under the respecitve root field set, for example `host.entity.*` , `user.entity.*`, etc.
+- If the entity type is not one of the above, then that `entity.*` fields should be nested under a new root-level object, called `generic`, as `generic.entity.*`
```

Comment:
Do you think `generic` will have any other fields apart from `generic.entity.*`

Reply:
Not foreseen.  Can anyone else think of a reason why we'd use the root field set `generic.*` for another purpose?